src/Microsoft.OpenApi/Services/OpenApiSecurityRequirementComparer.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. 
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.OpenApi.Models;
+
+namespace Microsoft.OpenApi.Services
+{
+    /// <summary>
+    /// Defines behavior for comparing properties of <see cref="OpenApiSecurityRequirement"/>.
+    /// </summary>
+    public class OpenApiSecurityRequirementComparer : OpenApiComparerBase<OpenApiSecurityRequirement>
+    {
+        /// <summary>
+        /// Executes comparision against source and target <see cref="OpenApiSecurityRequirement"/>.
+        /// </summary>
+        /// <param name="sourceSecurityRequirement">The source.</param>
+        /// <param name="targetSecurityRequirement">The target.</param>
+        /// <param name="comparisonContext">Context under which to compare the source and target.</param>
+        public override void Compare(
+            OpenApiSecurityRequirement sourceSecurityRequirement,
+            OpenApiSecurityRequirement targetSecurityRequirement,
+            ComparisonContext comparisonContext)
+        {
+            if (sourceSecurityRequirement == null && targetSecurityRequirement == null)
+            {
+                return;
+            }
+
+            if (sourceSecurityRequirement == null || targetSecurityRequirement == null)
+            {
+                comparisonContext.AddOpenApiDifference(
+                    new OpenApiDifference
+                    {
+                        OpenApiDifferenceOperation = OpenApiDifferenceOperation.Update,
+                        SourceValue = sourceSecurityRequirement,
+                        TargetValue = targetSecurityRequirement,
+                        OpenApiComparedElementType = typeof(OpenApiSecurityRequirement),
+                        Pointer = comparisonContext.PathString
+                    });
+
+                return;
+            }
+
+            var newSecuritySchemesInTarget = targetSecurityRequirement.Keys
+                .Where(targetReq => sourceSecurityRequirement.Keys.All(
+                    sourceReq => sourceReq.Reference.Id != targetReq.Reference.Id)).ToList();
+
+            foreach (var newSecuritySchemeInTarget in newSecuritySchemesInTarget)
+            {
+                WalkAndAddOpenApiDifference(
+                    comparisonContext,
+                    newSecuritySchemeInTarget.Reference.Id,
+                    new OpenApiDifference
+                    {
+                        OpenApiDifferenceOperation = OpenApiDifferenceOperation.Add,
+                        TargetValue = new KeyValuePair<OpenApiSecurityScheme, IList<string>>(
+                            newSecuritySchemeInTarget,
+                            targetSecurityRequirement[newSecuritySchemeInTarget]),
```

Comment:
This should not be a key-value pair. We are already pointing the the value itself - just use the IList<string>.